main.go

```diff
@@ -54,50 +61,58 @@ func main() {
 	}
 	log := zapr.NewLogger(zapLog)
 
-	err = run(log, args)
+	ctx := logr.NewContext(context.Background(), log)
+	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGTERM)
+	defer cancel()
+
+	err = run(ctx, args)
 	if err != nil {
 		log.Error(err, "")
 		os.Exit(1)
 	}
 	log.Info("gracefully shutdown")
 }
 
-func run(log logr.Logger, args *arguments) (err error) {
-	cs, err := pkgkubernetes.GetKubernetesClientset(args.KubeconfigPath)
+func run(ctx context.Context, args *arguments) error {
+	switch {
+	case args.Configuration != nil:
+		return configurationCommand(ctx, args.Configuration)
+	case args.Registry != nil:
+		return registryCommand(ctx, args.Registry)
+	}
+	return nil
```

Comment:
Return an error since this shouldn't happen.

Reply:
It would not make it past args parse but I added a default anyways.